pkg/controller/jobframework/setup.go

```diff
@@ -94,6 +89,43 @@ func SetupControllers(mgr ctrl.Manager, log logr.Logger, opts ...Option) error {
 	})
 }
 
+func setupControllerAndWebhook(mgr ctrl.Manager, gvk schema.GroupVersionKind, fwkNamePrefix string, cb IntegrationCallbacks, opts ...Option) error {
+    if err := cb.NewReconciler(
+        mgr.GetClient(),
+        mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", gvk.Kind, "managerName")), // Ensure managerName is defined or fetched
```

Comment:
I think "managerName" should still come from options
```suggestion
        options := ProcessOptions(opts...)
        mgr.GetEventRecorderFor(fmt.Sprintf("%s-%s-controller", gvk.Kind, options.ManagerName)), // Ensure managerName is defined or fetched
```